src/com/ms/silverking/log/Log.java

```diff
@@ -19,428 +18,445 @@
+    if (levelMet(Level.SEVERE)) {
+      try {
+        logQueue.put(new LogEntry(Level.SEVERE, m));
+      } catch (InterruptedException ie) {
+      }
+    }
+  }
+
+  public static void infoAsync(String m, Object o) {
+    if (levelMet(Level.INFO)) {
+      infoAsyncInternal(new LogEntry(Level.INFO, o, m));
+    }
+  }
+
+  public static void infoAsync(String m) {
+    if (levelMet(Level.INFO)) {
+      infoAsyncInternal(new LogEntry(Level.INFO, m));
+    }
+  }
+
+  public static void infoAsyncf(String f, Object... args) {
+    if (levelMet(Level.INFO)) {
+      infoAsyncInternal(new LogEntry(Level.INFO, String.format(f, args)));
+    }
+  }
+
+  private static void infoAsyncInternal(LogEntry le) {
+    try {
+      logQueue.put(le);
+    } catch (InterruptedException ie) {
+    }
+  }
+
+  public static void info(String m, Object o) {
+    if (levelMet(Level.INFO)) {
+      info(m + o);
+    }
+  }
+
+  public static void info(Object o) {
+    if (Log.levelMet(Level.INFO)) {
+      if (o != null) {
+        info(o.toString());
+      } else {
+        info("null");
+      }
+    }
+  }
+
+  public static void info(String m) {
+    if (Log.levelMet(Level.INFO)) {
+      log(Level.INFO, m);
+    }
+  }
+
+  public static void infof(String f, Object... args) {
+    if (levelMet(Level.INFO)) {
+      info(String.format(f, args));
+    }
+  }
+
+  public static void log(Object o) {
+    if (Log.levelMet(Level.INFO)) {
+      if (o != null) {
+        log(o.toString());
+      } else {
+        log("null");
+      }
+    }
+  }
+
+  public static void log(String m, Object o) {
+    if (levelMet(Level.INFO)) {
+      log(m + o);
+    }
+  }
+
+  public static void log(String m) {
+    log(Level.INFO, m);
+  }
+
+  private static void logError(Throwable e, String msg, Level l) {
+    logDest.logError(l, msg, e);
+  }
+
+  public static void logErrorWarning(Throwable e) {
+    logError(e, e.getMessage(), Level.WARNING);
+  }
+
+  public static void logErrorSevere(Throwable e, String className, String methodName) {
+    logErrorSevere(e, null, className, methodName);
+  }
+
+  public static void logErrorWarning(Throwable e, String msg) {
+    logError(e, msg, Level.WARNING);
+  }
+
+  public static void logErrorSevere(Throwable e, String msg, String className, String methodName) {
+    logError(e, msg, Level.SEVERE);
+  }
+
+  public static void fineAsync(String m, Object o) {
+    if (levelMet(Level.FINE)) {
+      fineAsyncInternal(new LogEntry(Level.FINE, o, m));
+    }
+  }
+
+  public static void fineAsync(Object o) {
+    if (levelMet(Level.FINE)) {
+      fineAsyncInternal(new LogEntry(Level.FINE, o.toString()));
+    }
+  }
+
+  public static void fineAsync(String m) {
+    if (levelMet(Level.FINE)) {
+      fineAsyncInternal(new LogEntry(Level.FINE, m));
+    }
+  }
+
+  private static void fineAsyncInternal(LogEntry le) {
+    try {
+      logQueue.put(le);
+    } catch (InterruptedException ie) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)